pathd/path_pcep_controller.c

```diff
@@ -454,13 +485,21 @@ int pcep_thread_timer_handler(struct thread *thread)
 
 	int ret = 0;
 	struct pcc_state *pcc_state = NULL;
-
+	pcc_state = get_pcc_state(ctrl_state, pcc_id);
+	assert(pcc_state != NULL);
 	switch (type) {
 	case TM_RECONNECT_PCC:
+		if (pcc_state->retry_count == 1) {
```

Comment:
You shouldn't use the pcc state here, it should be opaque, AFAIK the retry count was given by the pcc when calling the re-connection scheduling function.